Prepare for more DefaultFileSource changes

kkaefer opened the pull request.

jfirebaugh commented on test/storage/offline_database.test.cpp:
```diff
@@ -34,69 +35,60 @@ void deleteFile(const char* name) {
 
-TEST(OfflineDatabase, TEST_REQUIRES_WRITE(Create)) {
-    using namespace mbgl;
+class OfflineDatabaseTest : public ::testing::Test {
```
I dislike the "fixture" style of testing -- it makes it non-obvious what tests do, because they have hidden behavior without even an obvious function call you can trace through. Can we stick to the existing style?

jfirebaugh commented on src/mbgl/util/logging.cpp:
```diff
@@ -43,8 +43,7 @@ void Log::record(EventSeverity severity, Event event, int64_t code) {
 }
 
 void Log::record(EventSeverity severity, Event event, int64_t code, const std::string &msg) {
-    if (currentObserver && severity != EventSeverity::Debug &&
```
Was there a reason this was skipping `Debug` severity before?

kkaefer replied:
Not really; we aren't using Debug much. I changed the logging of one expected message from Debug to Info since it was a better fit anyway.

jfirebaugh commented on platform/default/sqlite3.cpp:
```diff
@@ -190,25 +187,46 @@ Database &Database::operator=(Database &&other) {
 
+
+DatabaseImpl::~DatabaseImpl() {
+    // Mark all transactions as invalid by removing the pointer to the database
```
When does this situation arise? Do we not want it to be considered a logic error, and require the database to outlive any transactions?

kkaefer commented:
Please re-review; I split up all of the commits to make them easier to review. I removed the common base class for tests, and they now explicitly perform all preparatory actions in each test body.

jfirebaugh commented on platform/default/mbgl/storage/offline_database.cpp:
```diff
@@ -28,67 +28,70 @@ OfflineDatabase::~OfflineDatabase() {
+        switch (userVersion()) {
+        case 0:
+        case 1:
+            // cache-only database; remove old table
```
```
// Newly created database, or old cache-only database; remove old table if it exists.
```

lilykaiser commented:
🎉
